_posts/2016-08-14-three-pillars-of-continuous-delivery.md

```diff
@@ -0,0 +1,47 @@
+
+### Automation, Automation, Automation
+
+**Automate the right thing.** It sounds very easy, but when I look now at some my past decisions I was not always automating the right part of my process. If you do your manual steps frequently enough or doing your manual tasks take a lot of time or your process is repeatable and complex, you have a good candidate for automation. Any automation without good context might lead you to improvement of unnecessary parts of your software delivery.
+
+**Use the right tools.** Identify the tools which can help you. In the modern software development we have plenty of options. Tools like Jenkins, Bamboo, Travis, Ansible, Docker can help you in the automation, but without a good justification of usage they can be just fancy tools. Your automation should evolved together with your needs and problems which are introduced by these problems. Proves of concept will allow you to understand if a specific tool is giving any value in the context of automation.
+
+### Rapid Response
+
+**Give answers now.** Continuous deployment should give your answer if your software is valid and ready to use. If your deployment process takes days, weeks or even hours in some cases, you can miss opportunity to get these answers. The time between the discovery and the fix of an issue can be minimised by the rapid response from your continuous deployment process.
+
+**Make it faster.**  Many different things can make your process slower. One of this problems is the organisation of your process. In many cases in my previous projects the original version of the process required the individual involvement of many people in the organisation. Sometimes you have to struggle with the slow access to the technical infrastructure in your company. Try to improve these element. The automation of your process should not only include the solutions for technical problems, but it should also improve non-technical aspects of the process.
+
+### Team Effort
+
+**Do it together.** In my opinion the responsibility for continuous delivery should be shared by the team. By the team I mean all people involved in the project. Not only software developers, not only system administrator, we should involve also business stakeholders. Understanding the continuous deployment by all team member strengthens knowledge about the software and improve cooperation. Our software is not longer a piece of lines of code, it is also place where this software lives.
```

Comment:
Our software is not longer a piece of lines of code....

the best I can suggest:

Our software is no longer just lines of code.....